### ISSUE TITLE
Feature request: Add flatpak support to Digidoc

### ISSUE THREAD

**Issue by Gren-95** [open-eid/DigiDoc4-Client#1299]
Hello,

Is it possible to build flatpak version of digidoc to simplify the installation on most Linux distros?

It would be simpler for the end user to just install it from flathub instead of building it from source.

**Comment by Gren-95** [comment]
I have recently found this project, but i would like to see it be incorporated in the main project if possible.

https://github.com/flathub/ee.ria.qdigidoc4

**Comment by ls-1N** [comment]
I agree - an official Flathub release would be more convenient, secure, and accessible. The RIA homepage can simply provide a Flathub link/button - more people are used to Google Play or Apple App Store links/buttons on websites instead of Windows .exe files directly on-site like in the olden days.

The advantage of an official Flathub release is that even though the community release is open-source, most users won't have the tech literacy to find out where their Linux app store (that has a Flathub integration) got the Digidoc4 app from. The same goes for actually understanding all the commits in a reasonable amount of time even for tech-savvier users.

So people will do the most convenient thing (use the app store) and not care about security if it is ambiguous and takes several extra steps. The extra maintenance overhead of a Flathub app is very low, I believe.

**Comment by ls-1N** [comment]
@Gren-95  In theory, #793 already has a discussion about packaging, including Snap, Appimage, and Flatpak.

**Comment by ls-1N** [comment]
> There is one main fundamental problem with these universal packaging formats. PCSCLite does not guarantee a stable API for smart card communication. At least three different versions are needed, and users must understand which OS supports which version (see [PCSCLite issue](https://github.com/LudovicRousseau/PCSC/issues/199), [qdigidoc commit](https://github.com/minosimo/qdigidoc-flatpak/commit/9195dc02f4ea13f9b9f11a7115c0b10b5b45cb6b)).
> 
> Other related issues:
> 
>     * MIME type registration (e.g., [Flatpak issue](https://github.com/minosimo/qdigidoc-flatpak/issues/21)).
> 
>     * AppImage lacks usable signing support for end users.
> 
>     * Some OSes require additional installation steps.
> 
> 
> Until these are resolved, I don't see them as viable mainline distribution methods. This is my personal opinion.

Maybe you and I should add our opinions/takes on #793 as well and close this particular issue since #793 has expanded in scope to also discuss Snap and Flatpak. Your counterpoint would calm some of the pings for updates there. As is now, issue #793 gives the _impression_ of being forgotten without any counterpoints or complexities. Ideally, the title on #793 should also be changed since some users read only the titles (not the whole issue discussion) to decide whether they should open a new issue (this one), thus this might happen again.

**Comment by apteryks** [comment]
Until flatkpaks are a thing, as of today there is also a `qdigidoc` package on GNU Guix, a functional package manager that can run atop any GNU/Linux distribution.
